pkg/controller/networkpolicy/validate.go

```diff
@@ -560,10 +560,18 @@ func (t *tierValidator) deleteValidate(oldObj interface{}, userInfo authenticati
 // validateAntreaGroupSelectors ensures that an IPBlock is not set along with namespaceSelector and/or a
 // podSelector.
 func validateAntreaGroupSelectors(s corev1a2.GroupSpec) (string, bool) {
+	podSelector, serviceRef, ipBlock := 0, 0, 0
+	if s.NamespaceSelector != nil || s.PodSelector != nil {
+		podSelector = 1
+	}
 	if s.IPBlock != nil {
-		if s.NamespaceSelector != nil && s.PodSelector != nil {
-			return fmt.Sprint("ClusterGroup IPBlock cannot be set with other selectors"), false
-		}
+		ipBlock = 1
+	}
+	if s.ServiceReference != nil {
+		serviceRef = 1
+	}
+	if podSelector+serviceRef+ipBlock > 1 {
+		return fmt.Sprint("ClusterGroup podSelector/namespaceSelector, serviceReference and ipBlock cannot be set with each other"), false
```

Comment:
suggestion: "At most one of podSelector/namespaceSelector, serviceReference or ipBlock can be set for a ClusterGroup"

BTW, we allow ClusterGroups without any selector?

Reply:
Good question... looking at `processSelector` I think if there's a CG w/o any selector at all, then it will simply have no members. If referred in a policy appliedToGroups and addressGroups will still be generated for it though. @abhiraut do you think we should add validation to ensure there's at least one selecting mechanism or did you not add that on purpose?

Reply:
I approved this PR as I don't think we need a resolution for this. It seems that the current behavior is ok (empty group with no members). In a way it is the same as using a Service reference for a Service that does not exist (yet?).

Reply:
intentionally left it out from validation to not do a GET on CGs and parse it every time an ACNP is created. Since it resolves to a group with no members, it does not fail open and maybe that's okay?